test/com/google/javascript/jscomp/Es6RewriteBlockScopedDeclarationTest.java

```diff
@@ -408,6 +408,28 @@ public void testLoopClosure() {
             "  })($jscomp$loop$2));",
             "  x++;",
             "}"));
+
+    // Preserve type annotation
+    test("for (;;){ /** @type {number} */ let x = 3; var f = function() { return x; } }",
+        LINE_JOINER.join(
+            "var $jscomp$loop$0 = {x: undefined};",
+            "for (;;$jscomp$loop$0 = {x: $jscomp$loop$0.x}) {",
+            "  /** @type {number} */ $jscomp$loop$0.x = 3;",
+            "  var f = function($jscomp$loop$0) {",
+            "    return function() { return $jscomp$loop$0.x}",
+            "  }($jscomp$loop$0);",
+            "}"));
+
+    // Preserve inline type annotation and constancy
+    test("for (;;){ const /** number */ x = 3; var f = function() { return x; } }",
```

Comment:
Let's add one more test where x is a "var" -- I just want to make sure we won't accidentally try to remove something from letConsts that isn't in it.